worker/workercmd/guardian.go

```diff
@@ -152,3 +177,64 @@ func detectGuardianFlags(logger lager.Logger) []string {
 func flagify(env string) string {
 	return strings.Replace(strings.ToLower(env), "_", "-", -1)
 }
+
+func getGdnFlagsFromConfig(configPath string) ([]string, error){
+	var configFlags []string
+
+	if configPath != "" {
+		file, err := os.Open(configPath)
+		if err != nil {
+			return []string{}, err
+		}
+		defer file.Close()
+		bs := bufio.NewScanner(file)
+		for bs.Scan() {
+			line := bs.Text()
+
+			if len(line) == 0 || line[0] == '#' || line[0] == ';' {
```

Comment:
One more thing I was thinking of is that it might not be the first character of the line that is a comment character, it can be indented or spaced in too. So maybe we should trim the spaces before we check for the comment character? Sorry I'm not sure if this is not really a reasonable concern though.

Reply:
Yeah, I think we should `TrimSpace` to match `go-flags`.

https://github.com/jessevdk/go-flags/blob/c17162fe8fd74f119ff938c5c67af63e3bac5ded/ini.go#L400

What's concerning to me is that we're only considering the `--long-flag` form of the value, whereas in `go-flags`, users can specify it in multiple ways. From the [godocs](https://godoc.org/github.com/jessevdk/go-flags#IniParser):

```
1. Compared to the ini-name tag on the option struct field (if present)
2. Compared to the struct field name
3. Compared to the option long name (if present)
4. Compared to the option short name (if present)
```

Is there any way we could `go-flags.IniParser` directly , rather than doing it ourselves?

Reply:
I hear your concerns and understand that string parsing when we are trying to extract values for functional purposes needs a lot of consideration. But the scenario here is different.

All we are trying to do is see _if the words "max-containers" and "network-pool" are present as the first argument on a line_. This is so we know to not supply our own default values for them. We aren't even concerned with what values they have in the file.

Given all that, comments and even other flags showing up in the list do not matter. `max-containers` and `network-pool` both in the structs here and in Guardian only have a "long" name listed in the CLI help. Therefore, the user should not be trying to supply any other types of names - ini, struct, short.

Reply:
That said, I'd prefer to use the IniParser as well but was not able to figure it out. I'm happy to take suggestions on how to use it but do not feel it's worth pursuing given the overall time the larger issue of containerd being blocked on BOSH has been taking.

Reply:
I think you're probably right that people won't use the `struct` version of the flags (and there are no `short` or `ini` versions).

One way that could possibly work (unverified) is to create a "mirror" of the [`GdnCommand` struct](https://github.com/cloudfoundry/guardian/blob/3837782d5e8fb8b7374e2b9109860690db15a5da/guardiancmd/command.go#L70) that contains only the fields we care about. e.g.

```go
type GdnCommand struct{
	ServerCommand  struct{
		Network struct{
			Pool string `long:"network-pool"`
		} `group:"Container Networking"

		Limits struct{
			MaxContainers *uint64 `long:"max-containers"`
		} `group:"Limits"
	} `command:"server"`
}
```

and then parsing into that struct:

```go
if *configFilePath != "" {
	var cmd GdnCommand
	parser := flags.NewParser(&cmd, flags.Default)
	parser.NamespaceDelimiter = "-"
	iniParser := flags.NewIniParser(parser)
	err := iniParser.ParseFile(*configFilePath)
	...
}
```

Reply:
Hm maybe @muntac if you don't have time, we could take over the PR and try to get it to use the iniParser?

Reply:
@clarafu sure, I'd appreciate that.

Reply:
Okay I pushed to the branch! If @muntac and @aoldershaw can take a look at it that would be great! I ran the acceptance testing described in the PR description against the changes and it seems to work fine, but it might be smart if you can take a quick look at it too.